src/main/java/de/bonndan/nivio/assessment/Assessable.java

```diff
@@ -0,0 +1,66 @@
+package de.bonndan.nivio.assessment;
+
+import de.bonndan.nivio.model.Component;
+import org.springframework.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static de.bonndan.nivio.assessment.StatusValue.SUMMARY_LABEL;
+
+/**
+ * Interface for components that can be assessed and can have assigned {@link StatusValue}s.
+ *
+ *
+ */
+public interface Assessable extends Component {
+
+    /**
+     * Returns the highest status as summary of all {@link StatusValue} and children summaries.
+     *
+     * @return status value, field contains the component identifier, message is the identifier of the highest status value
+     */
+    default StatusValue getOverallStatus() {
+
+        final AtomicReference<StatusValue> summary = new AtomicReference<>();
+        summary.set(new StatusValue(SUMMARY_LABEL, Status.UNKNOWN));
+
+        List<StatusValue> statusValues = new ArrayList<>(getStatusValues());
+        getChildren().forEach(o -> statusValues.add(o.getOverallStatus()));
+        statusValues.forEach(value -> {
+            if (value == null) {
+                return;
+            }
+
+            if (value.getStatus().isHigherThan(summary.get().getStatus())) {
+                summary.set(value);
+            }
+        });
+
+        return new StatusValue(SUMMARY_LABEL + "." + getIdentifier(), summary.get().getStatus(), summary.get().getField());
+    }
+
+    Set<StatusValue> getStatusValues();
+
+    default void setStatusValue(StatusValue statusValue) {
+
+        if (statusValue == null) {
+            throw new IllegalArgumentException("Status value is null");
+        }
+        if (StringUtils.isEmpty(statusValue.getField())) {
+            throw new IllegalArgumentException("Status value has no field");
```

Comment:
I think this check could also be moved to the ``StatusValue`` constructor. That would ensure that instances of that class are always valid.

Reply:
done